Cabal/Distribution/Package.hs

```diff
@@ -185,12 +211,117 @@ instance NFData ComponentId where
 
 -- | Returns library name prefixed with HS, suitable for filenames
 getHSLibraryName :: UnitId -> String
-getHSLibraryName (SimpleUnitId (ComponentId s)) = "HS" ++ s
+getHSLibraryName (UnitIdVar _) = error "getHSLibraryName: unbound variable"
+getHSLibraryName uid = "HS" ++ hashUnitId uid
 
--- | For now, there is no distinction between component IDs
--- and unit IDs in Cabal.
-newtype UnitId = SimpleUnitId ComponentId
-    deriving (Generic, Read, Show, Eq, Ord, Typeable, Data, Binary, Text, NFData)
+-- | An explicit substitution on modules.
+--
+-- NB: These substitutions are NOT idempotent, for example, a
+-- valid substitution is (A -> B, B -> A).
+type ModuleSubst = Map ModuleName Module
```

Comment:
I suggest we move all the utils for this type into the `BackPack` module, leaving just the type. We can leave a comment with a forwarding pointer to the utils that go with it.